server/main.py

```diff
@@ -0,0 +1,131 @@
+import json
+import os
+import uuid
+from datetime import datetime
+from flask import Flask, request, jsonify
+
+from utilities import dir_utils
+
+DEFAULT_PORT = 8080
+
+# create an instance of the Flask class and assigns it to the variable app
+app = Flask(__name__)
+# set the configuration parameter 'UPLOAD_FOLDER' in app object to the value of UPLOAD_FOLDER
+app.config['UPLOAD_FOLDER'] = dir_utils.UPLOAD_FOLDER
+
+
+def generate_filename(uid, original_filename):
+    """
+        Generate a new filename by combining the original filename, timestamp, and UID.
+
+        Args:
+            uid (str): Unique identifier for the file.
+            original_filename (str): Original filename of the uploaded file.
+
+        Returns:
+            str: New filename with the format 'original_filename_timestamp_uid'.
+    """
+    timestamp = datetime.now().strftime('%Y%m%d%H%M%S')
+    return f"{''.join(original_filename.split('.')[:-1])}_{timestamp}_{uid}.{original_filename.split('.')[-1]}"
```

Comment:
This line is messy, separate the different parts and give them meaningful names.